android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -270,21 +270,26 @@ class AddonsManagerAdapter(
         style?.maybeSetAddonNameTextColor(holder.titleView)
         style?.maybeSetAddonSummaryTextColor(holder.summaryView)
 
+        val statusErrorMessage = holder.statusErrorView.findViewById<TextView>(R.id.add_on_status_error_message)
+        val statusErrorLearnMoreLink = holder.statusErrorView.findViewById<TextView>(R.id.add_on_status_error_learn_more_link)
         if (addon.isDisabledAsBlocklisted()) {
-            holder.statusBlocklistedView.findViewById<TextView>(R.id.add_on_status_blocklisted_message).text =
-                context.getString(
-                    R.string.mozac_feature_addons_status_blocklisted,
-                    addonName,
-                )
-            holder.statusBlocklistedView.findViewById<TextView>(
-                R.id.add_on_status_blocklisted_learn_more_link,
-            ).setOnClickListener {
+            statusErrorMessage.text = context.getString(R.string.mozac_feature_addons_status_blocklisted, addonName)
+            statusErrorLearnMoreLink.setOnClickListener {
                 addonsManagerDelegate.onLearnMoreLinkClicked(
                     AddonsManagerAdapterDelegate.LearnMoreLinks.BLOCKLISTED_ADDON,
                     addon,
                 )
             }
-            holder.statusBlocklistedView.isVisible = true
+            holder.statusErrorView.isVisible = true
+        } else if (addon.isDisabledAsNotCorrectlySigned()) {
+            statusErrorMessage.text = context.getString(R.string.mozac_feature_addons_status_unsigned, addonName)
+            statusErrorLearnMoreLink.setOnClickListener {
+                addonsManagerDelegate.onLearnMoreLinkClicked(
+                    AddonsManagerAdapterDelegate.LearnMoreLinks.ADDON_NOT_CORRECTLY_SIGNED,
+                    addon,
+                )
+            }
+            holder.statusErrorView.isVisible = true
         }
```

Comment:
As there are more status coming we could even reuse the strings and error handling. But  I guess solutions will work at the end 😅 
  
```kotlin

        val (disabledErrorStringId, disabledStatus) = if (addon.isDisabledAsBlocklisted()) {
            R.string.mozac_feature_addons_status_blocklisted to AddonsManagerAdapterDelegate.LearnMoreLinks.BLOCKLISTED_ADDON
        } else if (addon.isDisabledAsNotCorrectlySigned()) {
            R.string.mozac_feature_addons_status_unsigned to AddonsManagerAdapterDelegate.LearnMoreLinks.ADDON_NOT_CORRECTLY_SIGNED
        } else null to null
        if (disabledErrorStringId != null && disabledStatus != null) {
            holder.statusUnsignedView.findViewById<TextView>(R.id.add_on_disabled_status_message).text =
                context.getString(
                    disabledErrorStringId,
                    addonName,
                )
            holder.statusUnsignedView.findViewById<TextView>(
                R.id.add_on_disabled_status_learn_more_link,
            ).setOnClickListener {
                addonsManagerDelegate.onLearnMoreLinkClicked(
                    disabledStatus,
                    addon,
                )
            }
            holder.statusUnsignedView.isVisible = true
        }
```

Reply:
as discussed on slack, I find the syntax a bit hard to understand so I'll keep it as it is for now

Reply:
👍🏽